### ISSUE TITLE
support to write arrow files to disk from other representations

### ISSUE THREAD

**Issue by behrica** [techascent/tech.ml.dataset#128]
For me the road for working with big data with t.d.m should look like this:

1. Taking the input data in any format and from any source
2. "convert" the data into a arrow file on local disk one time before analysis starts. This needs to happen in batches and the arrow file should be written in batches of a configurable size
3. use function read-stream-dataset-in-place to start accessing the data which stays on disk as mmaped files

Step 2. could be either done via a specific command line tool or we could have some functions here which makes this as easy as possible.

The arrow dependencies are already present in t.d.m, so is CSV support.
So the CSV -> arrow conversion should be the simplest.

The current function write-dataset-to-stream! is not enough, as it does not allow to work in batches , which is absolutely required for converting larger-then-RAM files.

Se we need to be able to read the data in blocks and write the blocks to disk, block-by-block.
The arrow java libraries allow this.
So does the univocity CVS parser

**Comment by behrica** [comment]
I googled quite intensively, but I did not find any standalone tool which converts "to arrow".
This is maybe not surprising,
as arrow was never positioned as a long term "data file storage" / data archive format.

So in this sense, it is the task of the analyst (so the user of t.m.d) to convert his data at a certain stage to arrow format.

This argument would support, that we should have this transformations supported by t.m.d
